otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py

```diff
@@ -0,0 +1,94 @@
+"""OTX DINO Class for mmdetection detectors."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import functools
+
+from mmdet.models.builder import DETECTORS
+
+from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
+    ActivationMapHook,
+    FeatureVectorHook,
+)
+from otx.algorithms.common.adapters.mmdeploy.utils import is_mmdeploy_enabled
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.detection.adapters.mmdet.models.detectors import CustomDeformableDETR
+
+logger = get_logger()
+
+
+@DETECTORS.register_module()
+class CustomDINO(CustomDeformableDETR):
+    """Custom DINO detector."""
+
+    def __init__(self, *args, task_adapt=None, **kwargs):
+        super().__init__(*args, task_adapt=task_adapt, **kwargs)
+        self._register_load_state_dict_pre_hook(
+            functools.partial(
+                self.load_state_dict_pre_hook,
+                self,
+            )
+        )
+
+    @staticmethod
+    def load_state_dict_pre_hook(model, ckpt_dict, *args, **kwargs):
+        """Modify mmdet3.x version's weights before weight loading."""
+
+        if list(ckpt_dict.keys())[0] == "level_embed":
+            logger.info("----------------- CustomDINO.load_state_dict_pre_hook() called")
+            # This ckpt_dict is come from mmdet3.x
```

Comment:
```suggestion
            # This ckpt_dict comes from mmdet3.x
```
Not important, but it seems typo.

Reply:
Fixed at 42d7e15